casper/src/main/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala

```diff
@@ -177,7 +177,7 @@ object RegistrySigGen {
         .withDeployer(ByteString.copyFrom(deployer))
         .withTimestamp(timestamp)
 
-    val rnd = Blake2b512Random(DeployData.toByteArray(seed))
+    val rnd = Blake2b512Random(DeployData.toByteArray(ProtoUtil.stripDeployData(seed)))
```

Comment:
The `seed` instantiated in line 175 contains only the deployer and timestamp fields. I cannot see any value in passing it to `stripDeployData` which isn't going to strip anything in this case, but return an identical copy of `seed`

Reply:
This aligns the RNG initialization with production code and makes the unforgable name inserted into the registry computed correctly. So the value, as the commit says, is fixing the RegistrySigGen tool

Reply:
This function generates an unforgeable name, it doesn't insert anything into the registry. 
Does `ProtoUtil.stripDeployData(seed)` return anything different than `seed` in this case?

Reply:
I don't understand the failure mode that's being fixed here either.

Reply:
Try singing a contract from scratch in a way that actually allows to resolve it from a the registry. You'll see that without this change, you need to pass to RegistriSigGen all of: timestamp, sk, uname. If you don't pass the uname, it's going to differ from what is actually registered, and the registry url provided by RegSigGen is going to be wrong.

Try with this change and uname is no longer needed.

Reply:
Sounds like a test that could be automated and should be a part of this PR

Reply:
I can confirm that Artur is right. The RegistrySigGen stopped working at some point.

Also I agree with Lukasz that we should automate its testing. It might be complicated to do so and I would be happy with just logging a JIRA issue